src/ng-select/lib/ng-select.component.ts

```diff
@@ -439,16 +445,24 @@ export class NgSelectComponent implements OnDestroy, OnChanges, AfterViewInit, C
             this.select(item);
         }
 
+        if (this._editableSearchTerm) {
+            this._setSearchTermFromItems();
+        }
+
         this._onSelectionChanged();
     }
 
     select(item: NgOption) {
         if (!item.selected) {
             this.itemsList.select(item);
-            if (this.clearSearchOnAdd) {
+            if (this.clearSearchOnAdd && !this._editableSearchTerm) {
                 this._clearSearch();
             }
 
+            if (this._editableSearchTerm) {
```

Comment:
does this really needed? because you are doing the same in method above `toggleItem` with:

```js
if (this._editableSearchTerm) {
            this._setSearchTermFromItems();
}
```